PathPlanning/BugPlanning/bug.py

```diff
@@ -0,0 +1,327 @@
+"""
+Bug Planning
+author: Sarim Mehdi(muhammadsarim.mehdi@studio.unibo.it)
+Source: https://sites.google.com/site/ece452bugalgorithms/
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+show_animation = True
+
+
+class BugPlanner:
+    def __init__(self, start_x, start_y, goal_x, goal_y, obs_x, obs_y):
+        self.goal_x = goal_x
+        self.goal_y = goal_y
+        self.obs_x = obs_x
+        self.obs_y = obs_y
+        self.r_x = [start_x]
+        self.r_y = [start_y]
+        self.out_x = []
+        self.out_y = []
+        for o_x, o_y in zip(obs_x, obs_y):
+            for add_x, add_y in zip([1, 0, -1, -1, -1, 0, 1, 1],
+                                    [1, 1, 1, 0, -1, -1, -1, 0]):
+                cand_x, cand_y = o_x+add_x, o_y+add_y
+                valid_point = True
+                for _x, _y in zip(obs_x, obs_y):
+                    if cand_x == _x and cand_y == _y:
+                        valid_point = False
+                        break
+                if valid_point:
+                    self.out_x.append(cand_x), self.out_y.append(cand_y)
+
+    def mov_normal(self):
+        return self.r_x[-1] + np.sign(self.goal_x - self.r_x[-1]), \
+               self.r_y[-1] + np.sign(self.goal_y - self.r_y[-1])
+
+    def mov_to_next_obs(self, visited_x, visited_y):
+        for add_x, add_y in zip([1, 0, -1, 0], [0, 1, 0, -1]):
+            c_x, c_y = self.r_x[-1] + add_x, self.r_y[-1] + add_y
+            for _x, _y in zip(self.out_x, self.out_y):
+                use_pt = True
+                if c_x == _x and c_y == _y:
+                    for v_x, v_y in zip(visited_x, visited_y):
+                        if c_x == v_x and c_y == v_y:
+                            use_pt = False
+                            break
+                    if use_pt:
+                        return c_x, c_y, False
+                if not use_pt:
+                    break
+        return self.r_x[-1], self.r_y[-1], True
+
+    '''Bug0: Greedy algorithm where you move towards goal
+    until you hit an obstacle. Then you go around it
+    (pick an arbitrary direction), until it is possible
+    for you to start moving towards goal in a greedy manner again'''
+    def bug0(self):
```

Comment:
Docstring usually is under function definition. Please check it 
https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard